concepts/operator-reconciliation/README.md

```diff
@@ -0,0 +1,150 @@
+
+The POCs main focus points should be
+
+1. Clarify which Operator Framework we can use and what Advantages / Disadvantages they have. We can leverage the knowledge of the team for this but should come up with a recommendation
+2. Decide a common approach to make sure that we are able to abstract the Operator communication not only for interaction with a local cluster (the cluster where the operator is running on) but also a remote cluster
+3. Make sure that we can safely watch all necessary Kyma Resources without much CPU or Memory Overhead for a Reconciler
+4. Make sure that our Developer Experience does not suffer from the new approach. Ideally, we want to be able to take over Component Reconciler Logic and keep the interfaces as stable as possible
+
+A derived action plan should tackle the following topics (from first to last):
+
+1. Create a decision document on which Operator Framework to use / not to use (e.g. kubebuilder, operator-sdk, native) and why (can be running document and updated regularly on changes)
+2. Spin-Up a Bootstrap Operator Repository for a Component Reconciler in which we can work further, communicate existance of POC goals in Kyma teams to raise awareness and prompt for collaborators
+3. Create an interface similar to our current K8s client in the reconciler that can deal with in-cluster and remote cluster API-Server Access
+4. Create an implementation that is able to deal with deploying and updating Helm Chart data (templated charts as native resources)
+5. Create a suggestion for a separate Provisioning Operator that can deploy CRDs and the Component Operator
+6. Derive a final Low-Level Architecture that suggests a scaled implementation while making sure to keep team responsibility separate
+
+## High level architecture
+
+For Operator-based reconciliations, we want to enable maximum flexibility of the architecture depending on the needs of the Kyma user. This means that we want to allow
+
+1. Fully Independent Small-Scale Clusters able to reconcile themselves, relying on external node setup (e.g. running for a local evaluation in k3d or for smaller scale deployments)
+2. Fully centralized Control Plane Configurations, in which the Control Plane is not only provisioning the cluster, but also hosting the reconciliation, ensuring minimal overhead in the workload cluster
+3. Lightweight Control Plane Configurations (Hybrid Mode), in which the Control Plane is able to provision the cluster and create/update an inventory of all managed cluster, but not taking care of reconciliation
+
+The main focus of the POC will be to prove the functionality of the Reconciliation Operator to work with all 3 modes independant of the final implementation. The POC needs to prove that we are able to reconcile a cluster remotely or in-cluster without a significant difference in reconciliation logic.
+
+![high_level_view_operator-reconciliation](assets/high_level_view_operator-reconciliation.png)
+
```

Comment:
Like the diagram .. but would be good to visualise also the Kyma CR (KEB is not talking to kyma-operator directly)

Would be great to add also a short table below the diagram describing each visualised component with 1-2 sentences what it is/what its purpose is

Reply:
Will take a bit but will do!

Reply:
FYI, we have a guideline for diagrams: https://kyma-project.io/community/guidelines/content-guidelines/09-diagrams/#examples

Reply:
@NHingerl  While I saw the guideline, I was using the Reference Diagramming language of SAP https://help.sap.com/docs/SAP_POWERDESIGNER/1cc460ad80f446e6a9d19303919ee269/c818cfa96e1b1014abb5d137d4620b1e.html?version=16.6.4&locale=en-US
http://www.fmc-modeling.org/download/fmc-and-tam/SAP-TAM_Standard.pdf

This is afaik the only widely accepted architecture standard @ SAP. If this is too far off for modelling kyma diagrams I think we will have to either rework these guidelines or add to them a mapping from TAM.

Reply:
As discussed, I think the SAP style guide you used is better suited to the needs of your document. 👍